### DISCUSSION TITLE
Should Zeek statically link external libraries?

### DISCUSSION THREAD

**Discussion by timwoj** [zeek/zeek#4213] [Archive]
This was brought up as part of https://github.com/zeek/cmake/pull/119. The question is whether Zeek should prefer static libraries from the OS vs shared libraries. For our bundled library builds, we tend to prefer static linking everywhere. Should it be the same for libraries we import from the OS?

**Comment by 0xxon** [comment]
My gut feeling is that, for operating system libraries, the answer should be to prefer dynamic libraries. That allows the OS to be responsible for the lifecycle management of the libraries they ship and, e.g., apply security updates to them.

One example where this can be important is OpenSSL.

If we link them statically, a rebuild of Zeek will be required for any updates of the OS libraries to be effective.
